drivers/flash/flash_stm32_ospi.c

```diff
@@ -519,14 +519,40 @@ static bool ospi_address_is_valid(const struct device *dev, off_t addr,
 	return (addr >= 0) && ((uint64_t)addr + (uint64_t)size <= flash_size);
 }
 
+static int stm32_ospi_wait_auto_polling(struct flash_stm32_ospi_data *dev_data,
+		OSPI_AutoPollingTypeDef *s_config, uint32_t timeout_ms)
+{
+	/* If interrupt is enabled, no need for busy waiting to check the poll result
+	 * like HAL_OSPI_AutoPolling() does.
+	 */
+	if (irq_is_enabled(DT_IRQN(STM32_OSPI_NODE))) {
```

Comment:
This being said I do see potential cases where irqs could be locked, like for instance if we're to support erase while execution in memory map mode while driver is relocated in RAM. Not supported yet, but under investigation.

In this case, we'd have to fallback on the HAL_OSPI_AutoPolling mode, don't we ?